### ISSUE TITLE
TypeScript statement completion with symbols does not always work 

### ISSUE THREAD

**Issue by leidegre** [microsoft/vscode#132683]
Issue Type: <b>Bug</b>

This is wonky.  I put together a small repo here https://gist.github.com/leidegre/e51afd311eb4d9729c3ea2a5e6411d3b

For example, here VS code is not giving me statement completion. I'm expecting to see `FOO` in the suggestion box.

![image](https://user-images.githubusercontent.com/63085/132568231-823a85a9-ca1e-4a48-8187-a0ef31b53be9.png)

but, if I move the symbol and type definition into a seperate file it does work

![image](https://user-images.githubusercontent.com/63085/132568362-279500ef-6c4c-4000-8e23-222ca1fff0d4.png)
![image](https://user-images.githubusercontent.com/63085/132568366-05fdf2b5-a679-4621-8272-47fec5ed716b.png)

but this isn't consistent. I have in a larger code examples of where it still doesn't work. I don't know why it works sometimes and sometimes not. it's functionally equivalent code.

also, apologies if this is better served by the TypeScript team but I thought I'd ask here first. I have observed this behavior for the past two years on both old and brand new systems. So it's been like this for a while but I'd like to fix that.

VS Code version: Code 1.60.0 (e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff, 2021-09-01T10:41:52.311Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 5950X 16-Core Processor             (32 x 3394)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.89GB (23.27GB free)|
|Process Argv|. --crash-reporter-id 888dbca9-2022-45aa-889b-64b30d1eaa60|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (29)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-mermaid|bie|1.11.0
mermaid-markdown-syntax-highlighting|bpr|1.2.2
prettier-vscode|esb|8.1.0
go|gol|0.27.2
reason-vscode|jar|1.7.13
jlang-clang|l3d|0.0.10
jlang-tundra2|l3d|0.0.1
vscode-azurefunctions|ms-|1.5.0
vscode-azureresourcegroups|ms-|0.4.0
vscode-azurestorage|ms-|0.12.1
csharp|ms-|1.23.15
python|ms-|2021.9.1191016588
vscode-pylance|ms-|2021.9.0
jupyter|ms-|2021.8.2031190714
jupyter-keymap|ms-|1.0.0
azure-account|ms-|0.9.8
cpptools|ms-|1.6.0
hexeditor|ms-|1.8.2
vsliveshare|ms-|1.0.4761
java|red|0.81.0
code-spell-checker|str|2.0.4
code-spell-checker-swedish|str|0.1.9
vscode-lua|tri|0.12.4
vscodeintellicode|Vis|1.2.14
vscode-java-debug|vsc|0.35.0
vscode-java-dependency|vsc|0.18.7
vscode-java-pack|vsc|0.18.4
vscode-java-test|vsc|0.31.3
vscode-maven|vsc|0.33.0

(1 theme extensions excluded)

</details>

**Comment by IllusionMH** [comment]
/needsMoreInfo

Is it reproducible with all extensions disabled?
You can try this with <kbd>F1</kbd> and `>Developer: Reload Window With Extensions Disabled`

Looks like in first case there are suggestions from IntelliCode extension visible first (stars before names) while in second case it's plain TS suggestions (no star before name)

![image](https://user-images.githubusercontent.com/1312662/132570482-d3d2a342-6c8e-4d5f-9355-11bc874a5404.png)
*(Screenshot from Untitled editor with TS 4.5, but shouldn't matter)*

You can see `FOO` as only suggestion si

P.S. Please post code samples not only in screenshots to help other reproduce your issues because typing code from screenshot isn't fun.

**Comment by leidegre** [comment]
No, it is not reproducible with extensions disabled. Though I have the same issue with the TypeScript playground and any embedded variant of Monaco that I have tested so far.

I'll see if I can narrow this down.

**Comment by leidegre** [comment]
> P.S. Please post code samples not only in screenshots to help other reproduce your issues because typing code from screenshot isn't fun.

There's a link to a gist with everything you need right there at the [top](https://gist.github.com/leidegre/e51afd311eb4d9729c3ea2a5e6411d3b).

**Comment by leidegre** [comment]
It does work as I expect, if I disable Visual Studio IntelliCode `visualstudioexptteam.vscodeintellicode`.

**Comment by leidegre** [comment]
I'll move this to the IntelliCode people, this is definitely begin caused by that plug-in.

It works if I disable the plug-in or if I use the following setting

```json
{
  "vsintellicode.typescript.completionsEnabled": true
}
```

And there's a odd blank suggestion there... probably something they need to take a look at... 

![image](https://user-images.githubusercontent.com/63085/132632270-7df2e9b0-83ad-4ad3-a3c9-0e7196963c86.png)

![image](https://user-images.githubusercontent.com/63085/132632303-98e618f3-29e2-43df-87fb-a7d2360d6a91.png)

**Comment by leidegre** [comment]
@IllusionMH do you know if this IntelliCode extension is installed by default? I'm kinda perplexed over why it has been installed the first place.

You can close this issue if you want. I just want an answer to that question. I'm going to create an issue with the IntelliCode people and reference this issue.

**Comment by IllusionMH** [comment]
> There's a link to a gist with everything you need right there at the top.

I'm sorry. My bad for missing it, thanks that you post everything and configs as well.

> Though I have the same issue with the TypeScript playground and any embedded variant of Monaco that I have tested so far.

Yep, can reproduce in TS Playground, but not in CodeSandbox (which uses parts of VS Code under the hood).
Probably former doesn't use tsserver as VS Code does or there are some other differences in approach to get completions.
Issue for Playground can be raised here https://github.com/microsoft/TypeScript-Website/issues

> do you know if this IntelliCode extension is installed by default? I'm kinda perplexed over why it has been installed the first place.

No, it's not installed by default. Clean VS Code installation comes with 0 extensions in "installed" section, and you can see all built-in extensions using `@builtin` filter in search field.

If you didn't install it manually then most likely as part of extension pack. [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) in your case.
